cpp/src/io/json/nested_json_gpu.cu

```diff
@@ -1611,7 +1634,9 @@ std::pair<std::unique_ptr<column>, std::vector<column_name_info>> json_column_to
                                                   mr);
 
       // Reset nullable if we do not have nulls
-      if (col->null_count() == 0) { col->set_null_mask(rmm::device_buffer{0, stream, mr}, 0); }
+      if (target_type.id() == type_id::STRING and col->null_count() == 0) {
+        col->set_null_mask(rmm::device_buffer{0, stream, mr}, 0);
+      }
```

Comment:
Can this be combined within the if statement below this?
```
      if (target_type.id() == type_id::STRING) {
        if (col->null_count() == 0) { col->set_null_mask(rmm::device_buffer{0, stream, mr}, 0); }
        return {std::move(col), {{"offsets"}, {"chars"}}};
      }
```
Non-string columns cannot have nulls then?

Reply:
Could you add this as a comment to the code? Or perhaps it is already commented somewhere else?

Reply:
Absolutely. Added a comment.